### PR TITLE
Implement quantized Attention on cpu

### PR DESCRIPTION
**Pull request by yufenglee**
**Description**: Describe your changes.
1. Implement quantized attention on CPU
2. Make both CPU and GPU support uni-direction
**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: onnxruntime/contrib_ops/cpu/quantization/attention_quant.cc [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+              hidden_size,                    // K      = NH
+              input_data + input_offset,      // A
+              hidden_size,                    // lda    = NH
+              input_zero_point,               // input zero point
+              weights_data + weights_offset,  // B
+              3 * hidden_size,                // ldb    = 3NH
+              weight_zero_point,              // weight zero point
+              qkv_dest + qkv_offset,          // C
+              head_size,                      // ldc
+              nullptr                         // use single-thread
```
Comment (tianleiwu) [inline onnxruntime/contrib_ops/cpu/quantization/attention_quant.cc:155]: Is there benefit using thread pool?

File: onnxruntime/contrib_ops/cpu/cpu_contrib_kernels.cc [hunk 2]
```diff
@@ -39,6 +39,8 @@ class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kMSDomain, 1,
 class ONNX_OPERATOR_KERNEL_CLASS_NAME(kCpuExecutionProvider, kMSDomain, 1, Gelu);
 class ONNX_OPERATOR_KERNEL_CLASS_NAME(kCpuExecutionProvider, kMSDomain, 1, BiasGelu);
 class ONNX_OPERATOR_KERNEL_CLASS_NAME(kCpuExecutionProvider, kMSDomain, 1, FastGelu);
+class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME( kCpuExecutionProvider, kMSDomain, 1, float_uint8_t_uint8_t, QAttention );
```
Comment (tianleiwu) [inline onnxruntime/contrib_ops/cpu/cpu_contrib_kernels.cc:42]: Format is different with other lines (extra space)

File: onnxruntime/python/tools/quantization/quantize.py [hunk 3]
```diff
@@ -1345,7 +1344,36 @@ def _quantize_matmul(self, node, new_nodes_list):
...
+        inputs.extend(quantized_input_names)
+        inputs.extend([node.input[2]])
+        inputs.extend(scale_names)
+        inputs.extend([node.input[3]])
+        inputs.extend(zero_point_names)
+
+        kwargs = {}
+        for attribute in node.attribute:
+            kwargs.update(_attribute_to_kwarg(attribute))
+        kwargs["domain"]=ms_domain
```
Comment (tianleiwu) [inline onnxruntime/python/tools/quantization/quantize.py:1372]: Format the script using ReformatPython.bat

File: onnxruntime/contrib_ops/cpu/quantization/attention_quant.cc (hunk 1 above)
Comment (yufenglee) [reply onnxruntime/contrib_ops/cpu/quantization/attention_quant.cc:155]: there is already a threadpool outside.

**Pull request merged**